www/js/search.js

```diff
@@ -726,76 +719,128 @@ module.exports = {
       });
   },
 
+  lineFactory(line, rows) {
+    const mainLineExists = !!document.querySelector('.main');
 
-  printShabad(rows, ShabadID, LineID) {
+    const shabadLine = h(
+      `li#li_${line.lineCount}`,
+      {
+        'data-line-count': line.lineCount,
+      },
+      h(
+        `a#line${line.ID}.panktee${
+           (line.mainLine && !mainLineExists) ? '.current.main.seen_check' : ''
+        }`,
+        {
+          'data-line-id': line.ID,
+          'data-main-letters': line.MainLetters,
+          onclick: e => this.clickShabad(e, line.ShabadID,
+                        line.ID, line, rows, 'click'),
+        },
+        [
+          h('i.fa.fa-fw.fa-check'),
+          h('i.fa.fa-fw.fa-home'),
+          ' ',
+          line.Gurmukhi,
+        ],
+      ),
+    );
+
+    return shabadLine;
+  },
+
+  printShabad(rows, ShabadID, LineID, start = 0) {
     const lineID = LineID || rows[0].ID;
     const shabadID = ShabadID || (rows[0].Shabads ? rows[0].Shabads[0].ShabadID : '');
-    let mainLine;
+    const lineIndex = rows.findIndex(row => row.ID === lineID);
     const shabad = this.$shabad;
     const apv = document.body.classList.contains('akhandpaatt');
 
+    // if the line clicked is further than throughput (in searches) then load until that line
+    let throughput = 40;
+
+    if (start) {
+      throughput = 10;
+    } else if (lineIndex > throughput) {
+      throughput = lineIndex + 1;
+    }
+
+    const end = start + throughput;
+
+    const lineHeight = 27;
```

Comment:
Can numbers like these be defined as constants somewhere? having them inline could make refactoring a pain.

Reply:
Yes, I agree. I would also like to break down this file into multiple files. I think we will address all these issues in technical debt sprint that's coming up.